disco_aws_automation/disco_deploy.py

```diff
@@ -743,30 +550,22 @@ def test_ami(self, ami, dry_run, deployment_strategy=None):
         hostclass = DiscoBake.ami_hostclass(ami)
         pipeline_hostclass_dict = self._hostclasses.get(hostclass)
         group = self._disco_group.get_existing_group(hostclass)
-        deployable = self.is_deployable(hostclass)
         testable = bool(self.get_integration_test(hostclass))
 
+        # We are only deployable in testing if we are in the pipeline. Otherwise assume that we aren't
+        # deployable. This must be done because we don't want to deploy things that shouldn't end up in
+        # the testing environment but do need to be tested. An example would be hostclasses that are
+        # only expected to exist in the bakery_environment. This doesn't hold true for update, which
+        # should just always deploy things unless explicitly told no.
+        deployable = pipeline_hostclass_dict and self.is_deployable(hostclass)
+
         if deployment_strategy is not None:
             desired_deployment_strategy = deployment_strategy
         else:
             desired_deployment_strategy = self.hostclass_option_default(hostclass, 'deployment_strategy',
```

Comment:
This seems kind of backwards. The strategy in `disco_aws.ini` is overriding the `--strategy` arg. I would have expected the reverse